paymentsheet/src/main/java/com/stripe/android/paymentsheet/navigation/PaymentSheetScreen.kt

```diff
@@ -56,6 +61,8 @@ internal sealed interface PaymentSheetScreen {
     val canNavigateBack: Boolean
     val sheetScreen: SheetScreen
 
+    fun title(isCompleteFlow: Boolean, isWalletEnabled: Boolean): StateFlow<ResolvableString?>
```

Comment:
Should we just return this & `showWalletsHeader` the actual values rather than have them wrapped as flows? We seem to be recreating the flows everytime we call this function in UI anyways.

Reply:
Oh nvm, they are dependent on the interactor state so we need it as a state flow. Anyway we can prevent recreating them on every render?

Reply:
Right now the easiest way to do this was to get the values for isCompleteFlow, isWalletEnabled from the UI, so we need it to be a function that returns a flow when those change.

We also need it to be a flow because some screens have internal state that cause the values to change.

Reply:
I can optimize this with remember. I've got a follow up ready after this PR is merged. Mind if I do the optimization as a follow up in the UI too?

Reply:
Yeah that's fine!

Reply:
https://github.com/stripe/stripe-android/pull/8801